supabase/migrations/20260807182308_canceled_org_app_retention_lifecycle.sql

```
@@ -0,0 +1,395 @@
+REVOKE ALL ON FUNCTION public.long_canceled_org_ids() FROM PUBLIC;
+REVOKE ALL ON FUNCTION public.long_canceled_org_ids() FROM anon;
+REVOKE ALL ON FUNCTION public.long_canceled_org_ids() FROM authenticated;
+GRANT ALL ON FUNCTION public.long_canceled_org_ids() TO service_role;
+
+COMMENT ON FUNCTION public.long_canceled_org_ids() IS
+  'Org ids whose stripe_info is canceled/deleted and GREATEST(canceled_at, subscription_anchor_end, trial_at) is older than 90 days.';
+
+-- Queue retention warning events (85d bundles / 90d apps).
+-- Window is [p_min_days, p_min_days+5) so warnings never fire in the same cron
+-- pass as the next destructive step (90d soft-delete / 95d app delete).
+-- Dedup: skip if already claimed in notifications OR still pending in the queue.
```

SQLFluff (4.2.2)

[error] Line is too long (81 > 80).

(LT05)